src/main/java/io/github/thebusybiscuit/slimefun4/core/machines/MachineOperation.java

```diff
@@ -57,4 +57,10 @@ default boolean isFinished() {
         return getRemainingTicks() <= 0;
     }
 
+    /**
+     * This method is called when a {@link MachineOperation} is interrupted before finishing.
+     * Implement to specify behaviour that should happen in this case.
+     */
+    default void cancel() {}
```

Comment:
Can we give this a better name? onBreak, onInterrupt, onCancel, I don't know. It's an event triggered function though, it doesn't cancel operation itself.

Reply:
renamed to ``onCancel``